### PR TITLE
dev-ml/camlp5: version bump to 7.10

### PR DESCRIPTION
**Pull request by donahue95**
bugs 704408, 704366, 644162 satisfied on my ~amd64 systems

### CODE REVIEW

**Comment by donahue95** [comment]
tagged with no signoff but 
Signed-off-by: Drake Allen Donahue <donahue95@comcast.net>
is in the commit messages.
The first time I used 
Signed-off-by: Drake Allen Donahue donahue95@comcast.net
the bot cleared no signoff then I read somewhere that <> should be used
What's Up?

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib findlib eutils
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: I don't see how any of these eclasses are being used in this ebuild.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+MY_P=${P%_p*}
+echo $MY_P
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: These are probably just for debugging purposes only, but you should take them out before pushing.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+DESCRIPTION="A preprocessor-pretty-printer of ocaml"
+HOMEPAGE="https://camlp5.github.io/"
+SRC_URI="https://github.com/camlp5/camlp5/archive/rel$(ver_rs 1- '').tar.gz -> ${P}.tar.gz"
+echo $SRC_URI
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: This too.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+S="${WORKDIR}/${PN}-rel$(ver_rs 1- '')"
+echo $S
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: And this.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+inherit multilib findlib eutils
+
+MY_P=${P%_p*}
+echo $MY_P
+DESCRIPTION="A preprocessor-pretty-printer of ocaml"
+HOMEPAGE="https://camlp5.github.io/"
+SRC_URI="https://github.com/camlp5/camlp5/archive/rel$(ver_rs 1- '').tar.gz -> ${P}.tar.gz"
+echo $SRC_URI
+S="${WORKDIR}/${PN}-rel$(ver_rs 1- '')"
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Please move S below RDEPEND (with one empty line in-between) so we keep good style consistency here.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+echo $S
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~alpha ~amd64 ~ppc ~x86"
+IUSE="doc +ocamlopt"
+
+DEPEND="
+	>=dev-lang/ocaml-3.10:=[ocamlopt?]
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: There are no ocaml-3* ebuilds in the tree, so this is always satisfied and can be lifted off. Place the USE flag condition to ocaml dependency below.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+PATCHLEVEL=${PV#*_p}
+PATCHLIST=""
+
+if [ "${PATCHLEVEL}" != "${PV}" ] ; then
+	for (( i=1; i<=PATCHLEVEL; i++ )) ; do
+		SRC_URI="${SRC_URI}
+			http://pauillac.inria.fr/~ddr/camlp5/distrib/src/patch-${PV%_p*}-${i} -> ${MY_P}-patch-${i}.patch"
+		PATCHLIST="${PATCHLIST} ${MY_P}-patch-${i}.patch"
+	done
+fi
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: What's going on in here?

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+src_prepare() {
+
+	for i in ${PATCHLIST} ; do
+		epatch "${DISTDIR}/${i}"
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Using `epatch` is banned in ebuilds. Use `eapply -p0` if you need 0 level.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+	done
+	eapply_user
+}
+
+src_configure() {
+	./configure \
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Why not `econf`, and why not use array form here?

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+		-bindir /usr/bin \
+		-libdir /usr/$(get_libdir)/ocaml \
+		-mandir /usr/share/man || die "configure failed"
+}
+
+src_compile(){
+	emake out
+	if use ocamlopt; then
+		emake  opt
+		emake  opt.opt
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Not that it (probably) breakes here, but there are two spaces.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	fi
+}
+
+src_install() {
+	emake DESTDIR="${D}" install
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Replace with `default`.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+	# findlib support
+	insinto "$(ocamlfind printconf destdir)/${PN}"
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Is this command from `findlib.eclass` or from ocaml package? Add `|| die` inside the subshell.

File: dev-ml/camlp5/camlp5-7.10.ebuild [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+	doins etc/META
+
+	use doc && dodoc -r doc/.
+	dodoc CHANGES DEVEL ICHANGES README.md UPGRADING MODE
```
Comment (juippis) [inline dev-ml/camlp5/camlp5-7.10.ebuild]: Add these to global `DOCS` array so they get installed with your `default` call in the beginning of this phase.

**Comment by donahue95** [comment]
juippis reviewed and said:
Also you have introduced 18 commits to modify 2 files. That's not okay. Please work in branches, and only make a single commit for this PR.
In my 3rd day of research I have yet to discover how my 1 commit became 18. Nor have I discovered a technique to eliminate the random commits and reduce this pull request to 1 commit. So I will attempt to close this pull request and start over.

**Comment by heroxbd** [comment]
Thanks you so much @donahue95, have you got the new PRs?

**Comment by donahue95** [comment]
Sorry, just a user who set out to solve camlp5 blocking my emerge -uND world. Thanks due to  juippis and atupone for trying to square me away and executing the merge.

**Comment by heroxbd** [comment]
> Sorry, just a user who set out to solve camlp5 blocking my emerge -uND world. Thanks due to juippis and atupone for trying to square me away and executing the merge.

No one is trying to square you away.  Learning to prepare pull requests in a clean way will benefit you in the long run way beyond Gentoo.

**Pull request closed without merging**